src/gsi/read_gnssrspd.f90

```diff
@@ -0,0 +1,515 @@
+     real(r_kind) :: dlat,dlon,dlat_earth,dlon_earth
+     real(r_kind) :: dlat_earth_deg,dlon_earth_deg
+     real(r_kind) :: cdist,disterr,disterrmax,rlon00,rlat00
+     real(r_kind) :: vdisterrmax,u00,v00,u0,v0
+     real(r_kind) :: dx,dy,dx1,dy1,w00,w10,w01,w11
+     real(r_kind) :: wdir,wspd
+     real(r_kind) :: tob,uob,vob,qob,spdob,rrob
+     real(r_kind) :: rhob,tdob
+     real(r_kind) :: pob_mb,pob_cb,pob_pa,gob
+     real(r_kind) :: psob_mb,psob_cb,psob_pa
+     real(r_kind) :: qmaxerr 
+     real(r_kind) :: dlnpsob,dlnpob,ppb
+     real(r_kind) :: crit1,timedif,xmesh,pmesh
+     real(r_kind) :: sstime,tdiff 
+     real(r_kind) :: tsavg,ff10,sfcr,zz
+     real(r_kind) :: es,qsat,rhob_calc,tdob_calc,tdry
+     real(r_kind) :: dummy 
+     real(r_kind) :: del,ediff,errmin,jbmin
+     real(r_kind) :: tvflg 
+
+     real(r_kind) :: presl(nsig)
+     real(r_kind) :: obstime(6,1)
+     real(r_kind) :: obsloc(2,1)
+     real(r_kind) :: obstmp(2,1)
+     real(r_kind) :: obswnd(4,1)
+     real(r_kind) :: obsfmr(2,1)
+     real(r_kind) :: obsmst(3,1)
+     real(r_kind) :: obsprs(1,1)
+     real(r_kind) :: obspsf(1,1)
+     real(r_kind) :: obsg10(1,1)
+     real(r_kind) :: obsqcm(2,1)
+     real(r_kind) :: gnssrw(2,1)
+      
+     real(r_double) :: rstation_id
+     real(r_double) :: r_prvstg(1,1),r_sprvstg(1,1)
+
+     real(r_kind), allocatable,dimension(:,:) :: cdata_all,cdata_out
+     real(r_kind), allocatable,dimension(:)   :: presl_thin
+
+!    Equivalence to handle character names
+     equivalence(r_prvstg(1,1),c_prvstg)
+     equivalence(r_sprvstg(1,1),c_sprvstg)
+     equivalence(rstation_id,c_station_id)
+
+!    Data 
+     data hdstr / 'SID' /
+     !data timestr  / 'YEAR MNTH DAYS HOUR MINU SECO' /
+     data timestr / 'DHR RPT' /
+     !data locstr   / 'CLAT CLON' /
+     data locstr   / 'XOB YOB' /
+     !data wndstr   / 'WSPD' / !GNSSRSPD Wind speed
+     data wndstr   / 'SOB' / !GNSSRSPD Wind speed
+     data oestr   / 'WSU' / !GNSSRSPD Wind speed uncertainty/error 
+     data lunin    / 13 /
+     data ithin    / -9 /
+     data rmesh    / -99.999_r_kind /
+ 
+!------------------------------------------------------------------------------------------------
+
+     write(6,*)'READ_GNSSRSPD: begin to read gnssrspd satellite data ...'
+
+!    Initialize parameters
+
+!    Set common variables
+     lspdob = obstype == 'gnssrspd'
+
+     nreal  = 0
+     iecol  = 0
+ 
+ 
+     lim_qm = 4
+     iecol=0
+     if (lspdob) then
+        nreal  = 23
+        iecol  =  4  
+        errmin = one
+     else 
+        write(6,*) ' illegal obs type in read_gnssrspd '
+        call stop2(94)
+     end if
+
+     inflate_error = .true.
+
+!    Check if the obs type specified in the convinfo is in the fl hdob bufr file 
+!    If found, get the index (nc) from the convinfo for the specified type
+     ntmatch =  0
+     ncsave  =  0
+     do nc = 1, nconvtype
+               if (trim(ioctype(nc)) == trim(obstype)) then 
```

Comment:
if blocks in this section of code need to be aligned properly

Reply:
Done!